storage/flux/table.gen.go

```diff
@@ -305,6 +306,370 @@ func (t *floatWindowTable) advance() bool {
 	return true
 }
 
+// window selector table
+type floatWindowSelectorTable struct {
+	floatTable
+	windowEvery int64
+	timeColumn  string
+}
+
+func newFloatWindowSelectorTable(
+	done chan struct{},
+	cur cursors.FloatArrayCursor,
+	bounds execute.Bounds,
+	every int64,
+	timeColumn string,
+	key flux.GroupKey,
+	cols []flux.ColMeta,
+	tags models.Tags,
+	defs [][]byte,
+	cache *tagsCache,
+	alloc *memory.Allocator,
+) *floatWindowSelectorTable {
+	t := &floatWindowSelectorTable{
+		floatTable: floatTable{
+			table: newTable(done, bounds, key, cols, defs, cache, alloc),
+			cur:   cur,
+		},
+		windowEvery: every,
+		timeColumn:  timeColumn,
+	}
+	t.readTags(tags)
+	t.init(t.advance)
+	return t
+}
+
+func (t *floatWindowSelectorTable) Do(f func(flux.ColReader) error) error {
+	return t.do(f, t.advance)
+}
+
+func (t *floatWindowSelectorTable) advance() bool {
+	arr := t.cur.Next()
+	if arr.Len() == 0 {
+		return false
+	}
+
+	cr := t.allocateBuffer(arr.Len())
+
+	switch t.timeColumn {
+	case execute.DefaultStartColLabel:
+		cr.cols[timeColIdx] = t.startTimes(arr)
+		t.appendBounds(cr)
+	case execute.DefaultStopColLabel:
+		cr.cols[timeColIdx] = t.stopTimes(arr)
+		t.appendBounds(cr)
+	default:
+		cr.cols[startColIdx] = t.startTimes(arr)
+		cr.cols[stopColIdx] = t.stopTimes(arr)
+		cr.cols[timeColIdx] = arrow.NewInt(arr.Timestamps, t.alloc)
+	}
+
+	cr.cols[valueColIdx] = t.toArrowBuffer(arr.Values)
+	t.appendTags(cr)
+	return true
+}
+
+func (t *floatWindowSelectorTable) startTimes(arr *cursors.FloatArray) *array.Int64 {
+	start := arrow.NewIntBuilder(t.alloc)
+	start.Reserve(arr.Len())
+
+	for _, v := range arr.Timestamps {
+		// modulo should produce non-negative remainder
+		remainder := v % t.windowEvery
+		if remainder < 0 {
```

Comment:
Exactly, before epoch.

Reply:
I opened this issue a while ago and maybe your change here is the fix: https://github.com/influxdata/flux/issues/2856

Reply:
Yes I didn't realize there was an issue for it. Yes I believe this change is the fix for that issue. I've added tests that will fail otherwise. I will check the aggregate code.